src/transformers/training_args.py

```diff
@@ -754,10 +760,16 @@ def __post_init__(self):
         if self.run_name is None:
             self.run_name = self.output_dir
 
-        if is_torch_available() and self.device.type != "cuda" and (self.fp16 or self.fp16_full_eval):
+        if is_torch_available() and self.device.type != "cuda" and (self.fp16 or self.fp16_full_eval or self.bfloat16):
+            raise ValueError(
+                "Mixed precision training with AMP or APEX (`--fp16`), FP16 evaluation, and bfloat16 can only be used on CUDA devices."
+            )
+
+        if (self.fp16 or self.fp16_full_eval or self.fp16_opt_level or self.fp16_backend) and self.bfloat16:
             raise ValueError(
-                "Mixed precision training with AMP or APEX (`--fp16`) and FP16 evaluation can only be used on CUDA devices."
+                "bfloat16 cannot be used together with AMP, APEX, or FP16."
```

Comment:
do you propose this until we sort out AMP/bf16 PR?

Reply:
Yes. I don't know whether AMP-bfloat16 will make it into transformers before this PR. If it does, we can adjust this check before closing the PR.